whittle/models/gpt/model.py

```diff
@@ -48,12 +48,14 @@ def __init__(self, config: Config) -> None:
         self.sub_network_intermediate_size = self.config.intermediate_size
         self.sub_network_num_heads = self.config.n_head
         self.sub_network_n_layers = self.config.n_layer
+        self.sub_network_head_size: int | None = self.config.head_size
+        self.sub_network_query_groups: int | None = self.config.n_query_groups
```

Comment:
Should it really be None now? It has to be present in the config and when setting the subnetwork, it will never be None

Reply:
the problem is that our Linter complains since it does some automated type checking later. Maybe there is a more elegant solution?